sys/kern/sysv_shm.c

```diff
@@ -561,22 +565,29 @@ kern_shmat_locked(struct thread *td, int shmid,
 	max_va = 0;
 #endif
 	vm_object_reference(shmseg->object);
-	rv = vm_map_find(&p->p_vmspace->vm_map, shmseg->object, 0, &attach_va,
+	attach_addr = attach_va;
```

Comment:
I don't understand this diff, `attach_va` isn't used again so why do we need to introduce a copy?

Reply:
It was done to differentiate the variable holding the hint and the one holding the actual capability, it doesn't add anything to the logic though, I'll take it out and reduce the diff

Reply:
shm requires some more work for reservations anyway, but this is probably not the PR for all of it.